src/EnergyPlus/DualDuct.cc

```diff
@@ -431,178 +394,164 @@ namespace DualDuct {
                                                                          cNumericFields);
 
                 // Anything below this line in this control block should use DDNum
-                DDNum = DamperIndex + state.dataDualDuct->NumDualDuctConstVolDampers;
+                DDNum = DamperIndex + NumDualDuctConstVolDampers;
+                auto &thisDD = state.dataDualDuct->dd_airterminal(DDNum);
                 GlobalNames::VerifyUniqueInterObjectName(
                     state, state.dataDualDuct->UniqueDualDuctAirTerminalNames, AlphArray(1), CurrentModuleObject, cAlphaFields(1), ErrorsFound);
-                state.dataDualDuct->dd_airterminal(DDNum).Name = AlphArray(1);
-                state.dataDualDuct->dd_airterminal(DDNum).DamperType = DualDuctDamper::VariableVolume;
-                state.dataDualDuct->dd_airterminal(DDNum).Schedule = AlphArray(2);
+                thisDD.Name = AlphArray(1);
+                thisDD.DamperType = DualDuctDamper::VariableVolume;
+                thisDD.Schedule = AlphArray(2);
```

Comment:
If you are immediately converting this to a `Sched"Ptr"`, do you even need to copy and save the `Schedule` name?